scripts/openmc-make-depletion-chain-casl

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env python3
+
+import glob
+import os
+from zipfile import ZipFile
+from collections import OrderedDict, defaultdict
+from io import StringIO
+from itertools import chain
+
+try:
+    import lxml.etree as ET
+    _have_lxml = True
+except ImportError:
+    import xml.etree.ElementTree as ET
+    _have_lxml = False
+
+import openmc.data
+import openmc.deplete
+from openmc._xml import clean_indentation
+from openmc.deplete.chain import _REACTIONS
+from openmc.deplete.nuclide import Nuclide, DecayTuple, ReactionTuple
+from openmc._utils import download
+
+from casl_chain import CASL_CHAIN
+
+URLS = [
+    'http://www.nndc.bnl.gov/endf/b7.1/zips/ENDF-B-VII.1-neutrons.zip',
+    'http://www.nndc.bnl.gov/endf/b7.1/zips/ENDF-B-VII.1-decay.zip',
+    'http://www.nndc.bnl.gov/endf/b7.1/zips/ENDF-B-VII.1-nfy.zip'
+]
+
+def main():
+    if os.path.isdir('./decay') and os.path.isdir('./nfy') and os.path.isdir('./neutrons'):
+        endf_dir = '.'
+    elif 'OPENMC_ENDF_DATA' in os.environ:
```

Comment:
Normally, it is best practice to have environment variable have higher precedence over filesystem settings.  The idea is, a shell user typically has much more control over environment variables than the filesystem.  Therefore, if a user sets an environment variable, that typically best indicates user intent.

For best practices managing environment state in python, I like to use the `boto` project (AWS's python API library) as a reference for best practices.  Here [1] is an example of their command line interface credentials preference chain, starting at arguments passed to a constructor (highest precedence) and ending with a default config file (lowest precedence):

[1] http://boto.cloudhackers.com/en/latest/boto_config_tut.html#credentials